homeassistant/components/swisscom/device_tracker.py

```diff
@@ -88,25 +96,30 @@ def get_swisscom_data(self):
         devices = {}
 
         try:
-            request = requests.post(url, headers=headers, data=data, timeout=10)
-        except (
-            requests.exceptions.ConnectionError,
-            requests.exceptions.Timeout,
-            requests.exceptions.ConnectTimeout,
-        ):
-            _LOGGER.info("No response from Swisscom Internet Box")
-            return devices
-
-        if "status" not in request.json():
-            _LOGGER.info("No status in response from Swisscom Internet Box")
-            return devices
-
-        for device in request.json()["status"]:
-            with suppress(KeyError, requests.exceptions.RequestException):
-                devices[device["Key"]] = {
-                    "ip": device["IPAddress"],
-                    "mac": device["PhysAddress"],
-                    "host": device["Name"],
-                    "status": device["Active"],
-                }
+            request = requests.post(
+                url, headers=headers, data=data, timeout=5, verify=self.verify_ssl
+            )
+        except requests.exceptions.ReadTimeout:
```

Comment:
This integration needs a 3rd party library for the interface code. We won't allow any major changes to the integration until that is done.

Please extract the device specific interface code and data parsing to a standalone library published on PyPI.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices